extensions/wikia/PhalanxII/classes/Phalanx.class.php

```diff
@@ -168,44 +171,87 @@ public function isTypeEmail() {
 	public function save() {
 		wfProfileIn( __METHOD__ );
 
-		if ( ( $this->data['type'] & self::TYPE_USER ) && User::isIP( $this->data['text'] ) ) {
-			if ( wfIsTrustedProxy( $this->data['text'] ) ) {
-				// Don't allow to set blocks for trusted proxies or Wikia network hosts
-				return false;
-			}
-
-			$this->data['ip_hex'] = IP::toHex( $this->data['text'] );
+		if ( !$this->validateUserFilter() ) {
+			return false;
 		}
 
-		$dbw = $this->getSharedDB( DB_MASTER );
+		$dbw = $this->getDatabase( DB_MASTER );
 		if ( empty( $this->data['id'] ) ) {
 			/* add block */
 			$dbw->insert( $this->db_table, $this->mapToDB(), __METHOD__ );
-			$action = 'add';
+			$action = static::ACTION_ADD_FILTER;
 		} else {
-			$dbw->update(
-				$this->db_table,
-				$this->mapToDB(),
-				[ 'p_id' => $this->data['id'] ],
-				__METHOD__
-			);
-			$action = 'edit';
+			$dbw->update( $this->db_table, $this->mapToDB(), array( 'p_id' => $this->data['id'] ), __METHOD__ );
+			$action = static::ACTION_MODIFY_FILTER;
 		};
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)